### ISSUE TITLE
Setting database configuration at run time

### ISSUE THREAD

**Issue by bnlambert** [actframework/actframework#342]
Bravo! man. This is a great job.
I wish to know if it is possible to configure and connect to database at run time, such that database config can be changed based on parameters from the request.
Also any recommendations/examples on how to document APIs written in Act Framework?

Thanks

**Comment by greenlaw110** [comment]
@bnlambert Here are few places you can refer to:

http://actframework.org/doc/model.md - general idea of using database including SQL and mongodb
https://github.com/actframework/act-sql-common - more details about SQL configuration
https://github.com/actframework/act-demo-apps - demo applications, find out those named with `todo` or `transaction` for database app demo
https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Java/act - TechEmpower benchmark project with ActFramework. You can find out how Act dealing with three databases with five database access technologies

**Comment by greenlaw110** [comment]
In terms of API, at the moment I don't have a web site for that. Please download them along with jar and source from maven repository

**Comment by greenlaw110** [comment]
Sorry @bnlambert I just relealized that you want to change database configuration at runtime. At the moment the framework doesn't support that. However nothing prevent the app from managing it's own datasource. The only thing is you can't use the built-in db access facilities, like Dao injection etc.

**Comment by bnlambert** [comment]
Thank you for the prompt response. Any idea on how to document APIs in Act?

**Comment by greenlaw110** [comment]
If you are talking about Swagger, at the moment there is no support for that ...

**Comment by bnlambert** [comment]
I mentioned swagger as an example but can some one document web APIs in act at the moment?

**Comment by greenlaw110** [comment]
No there is no tool to create your app's web API yet. It's actually a very interesting area that I want to spend my time. However it won't be available at least before Oct this year

**Comment by bnlambert** [comment]
I think one can already create Web API with Act, I am concern about the documentation. Never forget that you are doing a great Job. I will try to integrate some tools and see how it goes. Thank you very much for your support and courage.

**Comment by greenlaw110** [comment]
@bnlambert yes I mean generating Web API document for application. I look forward to seeing your work here. Cheer ...

**Comment by bnlambert** [comment]
Ok, Thanks
